src/EventHandler/Dispatcher.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\EventHandler;
+
+class Dispatcher
+{
+    private static array $listeners = [];
+    private static $instance = null;
+
+    private function __construct()
+    {
+    }
+
+    public static function getInstance()
+    {
+        if (self::$instance == null) {
+            self::$instance = new Dispatcher();
+        }
+
+        return self::$instance;
+    }
+
+    public function listen(string $event, callable $listener):void
+    {
+        self::$listeners[$event] = $listener;
```

Comment:
this will still only allow one listener per event... I think the method should be `addListener` in that case

Reply:
I will add the functionality for multiple listeners. @tidal suggested the method name as "listen".